core-common/src/main/java/org/zalando/nakadi/util/CompressionBodyRequestFilter.java

```diff
@@ -98,14 +104,31 @@ public BufferedReader getReader() throws IOException {
         }
     }
 
+    static class ZstdServletRequestWrapper extends HttpServletRequestWrapper {
```

Comment:
Seems that these classes (`GzipServletRequestWrapper` and `ZstdServletRequestWrapper`) are different only by the function that converts input streams one to each other. Does it make sense to simplify now to a single class with this function as a parameter? It will simplify addition of any other content-encodings/